0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,140 @@
+
+## Model of Operation
+
+### Negotiating the Pull Payment Agreement
+
+Prior to a Pull Payment, the Payer and the Payee have to negotiate a Pull Payment Agreement. This agreement SHOULD be comprised of the parameters defined in the `agreement`-object within the [Response Body](#Response-Body). The Payer's SPSP Server SHOULD store the negotiated Pull Payment Agreement parameters.
+
+The result of the negotiation is a Token representing the Pull Payment Agreement. The Token is required by the Payee's SPSP Client to connect to the Payer's SPSP server. It MUST be part of the [Payment Pointer](../0026-payment-pointers/0026-payment-pointers.md) and MUST therefore take any form that does not invalidate it.
+
+Implementations MAY try to parse the Token, however, it MUST support totally opaque tokens. 
+
+### Conducting the Pull Payment
+
+The Payee's SPSP Client opens a [STREAM](../0029-stream/0029-stream.md) connection to the Payer's SPSP Server as described in the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). Once this connection is established, the process continues as follows: 
+
+The SPSP Server begins sending ILP packets to complete the Pull Payment.
+  1. The SPSP Server will adjust their `sendMax` to reflect the amount they're willing to send.
+      * `sendMax` SHOULD be derived from the Pull Payment Agreement parameters.
+  2. The SPSP Client will adjust their `receiveMax` to reflect the amount they're willing to receive.
+      * `receiveMax` MAY be `Infinity`.
+  3. The SPSP Client's and Server's [STREAM Modules](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions) will move as much value as possible while staying inside these bounds.
+  4. If the SPSP Server reaches their `sendMax`, they end the stream and the connection. If the SPSP Client reaches their `receiveMax`, they will end the stream and the connection.
+
+The STREAM parameters - `sendMax` and `receiveMax` - are defined in [STREAM's frame encoding](../0029-stream/0029-stream.md#53-frames).
+
+## Specification
+
+### Query (`GET <SPSP Endpoint>`)
+
+The SPSP Client queries the SPSP Endpoint to get information about the SPSP Server:
+
+#### Request
+
+``` http
+GET /0f09dc92-84ad-401b-a7c9-441bc6173f4e HTTP/1.1
+Host: alice.com
+Accept: application/spsp4+json
+```
+Note that `0f09dc92-84ad-401b-a7c9-441bc6173f4e` is the Pull Payment Token.
+
+#### Response
+
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/spsp4+json
+
+{
+  "destination_account": "alice.ilpdemo.red~0f09dc92-84ad-401b-a7c9-441bc6173f4e",
+  "shared_secret": "k5nubgM6zpb88NPGVnI/tVjRdgpUh+JvMueRFEMvPcY=",
+  "balance": {
```

Comment:
how would you feel about putting all the fields (`balance`, `cycle`, and `agreement`) under a `pull` field? There's no field that says the type of payment this is, so someone might get confused deducing between an invoice and a pull payment if they both have balance fields or something

Reply:
I'm ok with doing that as long as we do the same for invoices, i.e. the balance moves within the invoice object.